stablehlo/tests/interpret_bitcast_convert.mlir

```diff
@@ -0,0 +1,224 @@
+                                        false, false, false, true,
+                                        true, true, true, false,
+                                        false, true, true, false,
+                                        true, false, true, false,
+                                        false, false, true, false,
+                                        true, true, false, false,
+                                        false, true, false, false,
+                                        true, false, false, false,
+                                        false, false, false, false]> : tensor<64xi1>
+  func.return
+}
+
+// -----
+
+func.func @bitcast_convert_op_test_f64_to_i4() {
+  %operand = stablehlo.constant dense<0x0123456789ABCDEF> : tensor<f64>
+  %result = "stablehlo.bitcast_convert"(%operand) : (tensor<f64>) -> tensor<16xi4>
+  check.expect_eq_const %result, dense<[0xF, 0xE, 0xD, 0xC,
+                                        0xB, 0xA, 0x9, 0x8,
+                                        0x7, 0x6, 0x5, 0x4,
+                                        0x3, 0x2, 0x1, 0x0]> : tensor<16xi4>
+  func.return
+}
+
+// -----
+
+func.func @bitcast_convert_op_test_f16_to_i64() {
+  %operand = stablehlo.constant dense<[0xCDEF, 0x89AB, 0x4567, 0x0123]> : tensor<4xf16>
+  %result = "stablehlo.bitcast_convert"(%operand) : (tensor<4xf16>) -> tensor<i64>
+  check.expect_eq_const %result, dense<0x0123456789ABCDEF> : tensor<i64>
+  func.return
+}
+
+// -----
+
+func.func @bitcast_convert_op_test_f64_to_f16() {
+  %operand = stablehlo.constant dense<0x0123456789ABCDEF> : tensor<f64>
+  %result = "stablehlo.bitcast_convert"(%operand) : (tensor<f64>) -> tensor<4xf16>
+  check.expect_almost_eq_const %result, dense<[0xCDEF, 0x89AB, 0x4567, 0x0123]> : tensor<4xf16>
+  func.return
+}
+
+// -----
+
+func.func @bitcast_convert_op_test_f64_to_f64() {
+  %operand = stablehlo.constant dense<0x0123456789ABCDEF> : tensor<f64>
+  %result = "stablehlo.bitcast_convert"(%operand) : (tensor<f64>) -> tensor<f64>
+  check.expect_almost_eq_const %result, dense<0x0123456789ABCDEF> : tensor<f64>
+  func.return
+}
+
+// -----
+
+func.func @bitcast_convert_op_test_c128_to_c64() {
+  %operand = stablehlo.constant dense<(0x0123456789ABCDEF, 0x0000000011111111)> : tensor<complex<f64>>
+  %result = "stablehlo.bitcast_convert"(%operand) : (tensor<complex<f64>>) -> tensor<2xcomplex<f32>>
+  check.expect_eq_const %result, dense<[(0x89ABCDEF, 0x01234567), (0x11111111, 0x00000000)]> : tensor<2xcomplex<f32>>
+  func.return
+}
+
+// -----
+
+func.func @bitcast_convert_op_test_c128_to_c128() {
+  %operand = stablehlo.constant dense<(0x0123456789ABCDEF, 0x0000000011111111)> : tensor<complex<f64>>
+  %result = "stablehlo.bitcast_convert"(%operand) : (tensor<complex<f64>>) -> tensor<complex<f64>>
+  check.expect_eq_const %result, dense<(0x0123456789ABCDEF, 0x0000000011111111)> : tensor<complex<f64>>
+  func.return
+}
+
+// -----
+
+func.func @bitcast_convert_op_test_c64_to_c128() {
```

Comment:
complex is not supported for this op right?

Reply:
Conversions between complex and non-complex types aren't supported but complex to complex is supported.